pysrc/pip_resolve.py

```diff
@@ -135,25 +135,47 @@ def create_package_as_root(package, dir_as_root):
             dir_as_root[DEPENDENCIES][package_as_root[NAME]] = package_tree
     return dir_as_root
 
-def satisfies_python_requirement(parsed_operator, py_version_str):
+
+def satisfies_python_requirement(parsed_operator, py_version):
+    """Check if a package required python versions matches the one of the system
+
+    Args:
+        parsed_operator (str): operator to compare by i.e. >, <=, ==
+        py_version (str): The python version that is required by the package
+
+    Returns:
+        bool: True if the version matches, False otherwise
+    """
     # TODO: use python semver library to compare versions
-    operator_func = {
+    operator = {
         ">": gt,
         "==": eq,
         "<": lt,
         "<=": le,
         ">=": ge,
         '!=': ne,
-    }[parsed_operator]
+    }
+    operator_func = operator.get(parsed_operator)
```

Comment:
Changed the way the operator function was established. It's pretty confusing to create the dictionary and extract values from it in one go. I think that is even more confusing for someone that doesn't work with Python on a daily basis.